### PR TITLE
Add a few isomip_plus configurations for wetting and drying

### PR DESCRIPTION
**Pull request by cbegeman**
This PR adds additional test cases with tidal forcing and a thin film for testing of the wetting and drying capability (i.e., grounding line motion). No changes to the documentation are needed. These tests are not added to the `wetdry` test suite; further testing of the wetting and drying scheme are needed to determine which `isomip_plus` tests are best suited for this purpose.

* [X] Document (in a comment titled `Testing` in this PR) any testing that was used to verify the changes

### CODE REVIEW

**Comment by cbegeman** [comment]
@xylar Here I'm adding a few additional tests that I used to investigate some of the issues with RK4 and wetting and drying. I'm out this week so no rush on the review.

**Review by xylar — COMMENTED** [review]
I'm seeing the following traceback when I run the `ocean/isomip_plus/planar/2km/z-star/Ocean0` test case with this branch:
```
Traceback (most recent call last):
  File "/gpfs/fs1/home/ac.xylar/compass/enhance-isomip-plus/compass/run/serial.py", line 320, in _log_and_run_test
    _run_test(test_case, available_resources)
  File "/gpfs/fs1/home/ac.xylar/compass/enhance-isomip-plus/compass/run/serial.py", line 417, in _run_test
    _run_step(test_case, step, test_case.new_step_log_file,
  File "/gpfs/fs1/home/ac.xylar/compass/enhance-isomip-plus/compass/run/serial.py", line 468, in _run_step
    step.run()
  File "/gpfs/fs1/home/ac.xylar/compass/enhance-isomip-plus/compass/ocean/tests/isomip_plus/initial_state.py", line 87, in run
    self._plot(ds)
  File "/gpfs/fs1/home/ac.xylar/compass/enhance-isomip-plus/compass/ocean/tests/isomip_plus/initial_state.py", line 294, in _plot
    plotter.plot_3d_field_top_bot_section(
  File "/gpfs/fs1/home/ac.xylar/compass/enhance-isomip-plus/compass/ocean/tests/isomip_plus/viz/plot.py", line 597, in plot_3d_field_top_bot_section
    self.plot_horiz_series(daTop,
  File "/gpfs/fs1/home/ac.xylar/compass/enhance-isomip-plus/compass/ocean/tests/isomip_plus/viz/plot.py", line 532, in plot_horiz_series
    self._plot_horiz_field(field, title=title, outFileName=outFileName,
  File "/gpfs/fs1/home/ac.xylar/compass/enhance-isomip-plus/compass/ocean/tests/isomip_plus/viz/plot.py", line 821, in _plot_horiz_field
    plt.colorbar(localPatches, extend='both')
  File "/gpfs/fs1/home/ac.xylar/chrysalis/mambaforge/envs/compass_test/lib/python3.10/site-packages/matplotlib/pyplot.py", line 2137, in colorbar
    ret = gcf().colorbar(mappable, cax=cax, ax=ax, **kwargs)
  File "/gpfs/fs1/home/ac.xylar/chrysalis/mambaforge/envs/compass_test/lib/python3.10/site-packages/matplotlib/figure.py", line 1310, in colorbar
    cb = cbar.Colorbar(cax, mappable, **cb_kw)
  File "/gpfs/fs1/home/ac.xylar/chrysalis/mambaforge/envs/compass_test/lib/python3.10/site-packages/matplotlib/_api/deprecation.py", line 384, in wrapper
    return func(*inner_args, **inner_kwargs)
  File "/gpfs/fs1/home/ac.xylar/chrysalis/mambaforge/envs/compass_test/lib/python3.10/site-packages/matplotlib/colorbar.py", line 418, in __init__
    self._draw_all()
  File "/gpfs/fs1/home/ac.xylar/chrysalis/mambaforge/envs/compass_test/lib/python3.10/site-packages/matplotlib/colorbar.py", line 579, in _draw_all
    self._add_solids(X, Y, self._values[ind, np.newaxis])
  File "/gpfs/fs1/home/ac.xylar/chrysalis/mambaforge/envs/compass_test/lib/python3.10/site-packages/matplotlib/colorbar.py", line 595, in _add_solids
    self.solids = self.ax.pcolormesh(
  File "/gpfs/fs1/home/ac.xylar/chrysalis/mambaforge/envs/compass_test/lib/python3.10/site-packages/matplotlib/__init__.py", line 1446, in inner
    return func(ax, *map(sanitize_sequence, args), **kwargs)
  File "/gpfs/fs1/home/ac.xylar/chrysalis/mambaforge/envs/compass_test/lib/python3.10/site-packages/matplotlib/axes/_axes.py", line 6227, in pcolormesh
    collection = mcoll.QuadMesh(
  File "/gpfs/fs1/home/ac.xylar/chrysalis/mambaforge/envs/compass_test/lib/python3.10/site-packages/matplotlib/collections.py", line 1939, in __init__
    super().__init__(**kwargs)
  File "/gpfs/fs1/home/ac.xylar/chrysalis/mambaforge/envs/compass_test/lib/python3.10/site-packages/matplotlib/_api/deprecation.py", line 454, in wrapper
    return func(*args, **kwargs)
  File "/gpfs/fs1/home/ac.xylar/chrysalis/mambaforge/envs/compass_test/lib/python3.10/site-packages/matplotlib/collections.py", line 157, in __init__
    cm.ScalarMappable.__init__(self, norm, cmap)
  File "/gpfs/fs1/home/ac.xylar/chrysalis/mambaforge/envs/compass_test/lib/python3.10/site-packages/matplotlib/cm.py", line 398, in __init__
    self.set_norm(norm)  # The Normalize instance of this ScalarMappable.
  File "/gpfs/fs1/home/ac.xylar/chrysalis/mambaforge/envs/compass_test/lib/python3.10/site-packages/matplotlib/cm.py", line 636, in set_norm
    self.norm = norm
  File "/gpfs/fs1/home/ac.xylar/chrysalis/mambaforge/envs/compass_test/lib/python3.10/site-packages/matplotlib/cm.py", line 617, in norm
    self._id_norm = self.norm.callbacks.connect('changed',
  File "/gpfs/fs1/home/ac.xylar/chrysalis/mambaforge/envs/compass_test/lib/python3.10/site-packages/matplotlib/cbook/__init__.py", line 232, in connect
    if proxy in self._func_cid_map[signal]:
KeyError: 'changed'
```

The following run fine:
```
ocean/isomip_plus/planar/2km/sigma/thin_film_tidal_forcing_Ocean0
ocean/isomip_plus/planar/2km/single_layer/tidal_forcing_Ocean0
ocean/isomip_plus/planar/2km/single_layer/thin_film_tidal_forcing_Ocean0
```

I will look into whether this is something on `main` or something introduced here.

In the meantime, I have a few small suggestions.

**Comment by xylar** [comment]
@cbegeman, could you add back the `Testing` item in the checklist?
https://github.com/MPAS-Dev/compass/blob/main/.github/pull_request_template.md?plain=1#L17
I think that item should apply (and is still needed) even if the other check boxes don't.

**Comment by xylar** [comment]
I can confirm that `ocean/isomip_plus/planar/2km/z-star/Ocean0` runs fine on `main` so the issue above must have somehow been introduced here.

File: compass/ocean/tests/isomip_plus/forward.py [hunk 1]
```diff
@@ -145,6 +142,22 @@ def run(self):
...
+        dt_btr_per_km = config.getfloat('isomip_plus', 'dt_btr_per_km')
+
+        # https://stackoverflow.com/a/1384565/7728169
+        # Note: this will drop any fractional seconds, which is usually okay
+        dt = time.strftime('%H:%M:%S', time.gmtime(dt_per_km * resolution))
+        btr_dt = time.strftime(
+            '%H:%M:%S', time.gmtime(dt_btr_per_km * resolution))
+
+        options = dict(config_dt="'{}'".format(dt),
+                       config_btr_dt="'{}'".format(btr_dt))
```
Comment (xylar) [inline compass/ocean/tests/isomip_plus/forward.py:158]: Let's stick with f-strings:
```suggestion
        options = dict(config_dt=f"'{dt}'",
                       config_btr_dt=f"'{btr_dt}'")
```

**Comment by xylar** [comment]
Never mind, I recreated my `compass_test` environment and reran `ocean/isomip_plus/planar/2km/z-star/Ocean0` successfully.  Whatever when wrong, I think it was something strange I did.

**Review by xylar — APPROVED** [review]
Looks great!  Merge when you're ready.

**Pull request merged**